src/main.cpp

```diff
@@ -21,5 +22,17 @@ int main(int argc, char** argv)
         return 1;
     }
 
-    return app.run(*mainWindow);
+    int retCode = 0;
+    if(Settings::instance().startAtTray())
+    {
+        mainWindow->hide();
+        Application::instance().keepAlive();
+        retCode = app.run();
+    }
+    else
+    {
+        retCode = app.run(*mainWindow);
```

Comment:
If possible run this first without the if/else branch and then do the things in the if branch to make it simpler. Something like this;
```
auto const retCode = app.run(*mainWindow);
if(Settings::instance().startAtTray())
{
    mainWindow->hide();
    Application::instance().keepAlive();
}

return retCode;
```

Reply:
I have tried this approach. But it didn't work